src/adapter/src/catalog.rs

```diff
@@ -6744,6 +6745,21 @@ impl Catalog {
                 ),
                 tcp_user_timeout: Some(self.system_config().pg_replication_tcp_user_timeout()),
             },
+            upsert_rocksdb_tuning_config: Some({
+                let config_str = self.system_config().upsert_rocksdb_tuning_config();
+                match serde_json::from_str(config_str) {
```

Comment:
I'm not sure if the `serde_json::from_str` should not be moved into a dedicated `impl Value for RocksDBTuningParameters`. The benefit would be that malformed values will be reported immediately. With the current setup if somebody sets `upsert_rocksdb_tuning_config` to a bad value (but a valid JSON):

- the change will be accepted by the LD sync loop, but
- upon restart the all environments that accepted that change will emit the warning.

Reply:
How are malformed values reported if the value deserialization fails? I wanted to ensure we make forward progress, and just let someone see the sentry warning later on

Reply:
@aalexandrov one thing we can do is:

```
struct SerdeJson<T>(pub T)

impl<T: serde::Deserialize> Value for SerdeJson<T> { ... }
```

However, this would mean we would need to add `rocksdb` as a dep of `sql`, so we can access to the tuning parameter struct, which I was trying to avoid